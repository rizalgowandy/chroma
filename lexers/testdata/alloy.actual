sig Node {
  edges: set Node
}

fact "Connected graph" {
  some n: Node | n.*edges = Node
}

fact "No self edges" {
  no iden & edges
}

one sig Ball {
  -- note the var
  var loc: Node
}
