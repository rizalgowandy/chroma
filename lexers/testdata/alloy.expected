[
  {"type":"KeywordDeclaration","value":"sig"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Node"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Name","value":"edges"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Keyword","value":"set"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Node"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"fact"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"\"Connected graph\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Keyword","value":"some"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"n"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Node"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"|"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"n"},
  {"type":"Operator","value":".*"},
  {"type":"Name","value":"edges"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Node"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"fact"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralString","value":"\"No self edges\""},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Keyword","value":"no"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordConstant","value":"iden"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"\u0026"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"edges"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"Keyword","value":"one"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordDeclaration","value":"sig"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Ball"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"{"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"CommentSingle","value":"-- note the var"},
  {"type":"TextWhitespace","value":"\n  "},
  {"type":"Keyword","value":"var"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"loc"},
  {"type":"Punctuation","value":":"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"Node"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Operator","value":"}"},
  {"type":"TextWhitespace","value":"\n"}
]
